subprojects/core-api/src/main/java/org/gradle/api/provider/HasConfigurableValue.java

```diff
@@ -43,14 +43,31 @@
      *
      * <p>Note that although the value of this object will no longer change, the value may itself be mutable.
      * Calling this method does not guarantee that the value will become immutable, though some implementations may support this.</p>
+     *
+     * <p>If the value of this object is already final, this method does nothing.</p>
      */
     void finalizeValue();
 
+    /**
+     * Requests that the final value of this object be calculated on the next read of the value, if not already known.
```

Comment:
I think it would be nice to clarify the interaction with `disallowChanges`. In particular it's unclear to me if it _implies_ that further changes are going to be ignored _or_ that they would trigger an error if they happen.